stglib/rsk/rsk2cdf.py

```diff
@@ -41,49 +41,37 @@ def rsk_to_xr(metadata):
     ds = utils.write_metadata(ds, metadata)
 
     print(('Loading from sqlite file %s; '
-           'this may take a while for large datasets') % rskfile)
-
-    conn = init_connection(rskfile)
-
-    conn.execute("SELECT tstamp, channel01 FROM burstdata")
-    data = conn.fetchall()
-    print("Done fetching data")
-    d = np.asarray(data)
-    # Get samples per burst
-    try:
-        # this seems to be used on older-style databases;
-        # throws error on newer files
-        samplingcount = conn.execute(
-            "select samplingcount from schedules").fetchall()[0][0]
-    except sqlite3.OperationalError:
-        samplingcount = conn.execute(
-            "select samplingcount from wave").fetchall()[0][0]
-    ds.attrs['samples_per_burst'] = samplingcount
-
-    try:
-        samplingperiod = conn.execute(
-            "select samplingperiod from schedules").fetchall()[0][0]
-    except sqlite3.OperationalError:
-        samplingperiod = conn.execute(
-            "select samplingperiod from wave").fetchall()[0][0]
-    ds.attrs['sample_interval'] = samplingperiod / 1000
-
-    try:
-        repetitionperiod = conn.execute(
-            "select repetitionperiod from schedules").fetchall()[0][0]
-    except sqlite3.OperationalError:
-        repetitionperiod = conn.execute(
-            "select repetitionperiod from wave").fetchall()[0][0]
-    ds.attrs['burst_interval'] = repetitionperiod / 1000
-
-    ds.attrs['burst_length'] = ds.attrs['samples_per_burst'] * \
-        ds.attrs['sample_interval']
-    ds.attrs['serial_number'] = str(conn.execute(
-        "select serialID from instruments").fetchall()[0][0])
-    ds.attrs['INST_TYPE'] = 'RBR Virtuoso d|wave'
-
-    conn.close()
-
+           'this may take a while for large datasets') % rskfile)    
+
+     #Assume RBRvirtuoso in burst mode if no attrs
+    if 'instrument_type' not in ds.attrs.keys():
```

Comment:
I prefer this logic without `.keys()`: e.g., `if 'instrument_type' not in ds.attrs`, here and elsewhere

Reply:
changes